test/server.test.js

```diff
@@ -1813,6 +1813,23 @@ test('gh-779 set-cookie fields should never have commas', function (t) {
 });
 
 
+test('gh-986 content-type fields should never have commas', function (t) {
+    SERVER.get('/content-type', function (req, res, next) {
+        res.header('content-type', 'foo');
+        res.header('content-type', 'bar');
+        res.send(200);
+    });
+
+    CLIENT.get('/content-type', function (err, _, res) {
+        t.ifError(err);
+        t.equal(Array.isArray(res.headers['content-type']), false,
+                'content-type header should not be an array');
+        t.equal(res.headers['content-type'], 'bar');
+        t.end();
+    });
+});
+
+
```

Comment:
Can you add another testcase using `res.setHeader`, which should have the same functionality, but just to cover both cases?